File: src/main/java/com/bokeunjeong/portfolio/repository/PortfolioProjectRepository.java
package com.bokeunjeong.portfolio.repository;

import com.bokeunjeong.portfolio.model.entity.Project;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PortfolioProjectRepository extends JpaRepository<Project, String> {

    List<Project> findAllByDisplayOrderByIdDesc(Boolean display);

    List<Project> findAllByTypeOrderByIdDesc(String type);

}

File: src/main/kotlin/com/bokeunjeong/portfolio/repository/PortfolioProjectRepository.kt
package com.bokeunjeong.portfolio.repository

import com.bokeunjeong.portfolio.dto.PortfolioProjectDto
import org.springframework.data.jpa.repository.JpaRepository


interface PortfolioProjectRepository : JpaRepository<PortfolioProjectDto, Int> {
    fun findByName(name: String): List<PortfolioProjectDto>

    fun findTop5ByOrderByIdDesc(): List<PortfolioProjectDto>
}